### PR TITLE
Added permissions to resources for metrics exporter

### PR DESCRIPTION
**Pull request by aruniiird**
Remove an accidental role permissions added to prometheus.
Add the same (missing) permissions to metrics exporter.

A minor code refactor to optimize role/rolebinding addition to metrics exporter
and a minor bug fix

### CODE REVIEW

**Comment by aruniiird** [comment]
@umangachapagain , we are not seeing any `*v1.StorageClass` errors anymore.
Fixing other permission issues as seen in the ocs-metrics-exporter logs.
Please take a look.

**Comment by aruniiird** [comment]
Still seeing error messages, like the following,
```
failed to get PVC appset1/busybox-pvc: persistentvolumeclaims "busybox-pvc" is forbidden:
User "system:serviceaccount:openshift-storage:ocs-metrics-exporter" cannot get resource "persistentvolumeclaims" in API group "" in the namespace "appset1"
```
This is an expected error message, as we are granting `ocs-metrics-exporter` permissions to watch only the namespace it is currently deployed (here it is `openshift-storage`).

Question here is, above error message popped up in a MCO cluster and is this an expected behavior in this type of scenario?

**Comment by umangachapagain** [comment]
> Still seeing error messages, like the following,
> 
> ```
> failed to get PVC appset1/busybox-pvc: persistentvolumeclaims "busybox-pvc" is forbidden:
> User "system:serviceaccount:openshift-storage:ocs-metrics-exporter" cannot get resource "persistentvolumeclaims" in API group "" in the namespace "appset1"
> ```
> 
>
> This is an expected error message, as we are granting `ocs-metrics-exporter` permissions to watch only the namespace it is currently deployed (here it is `openshift-storage`).
>
This shouldn't happen. There's a bug somewhere.
> 
> Question here is, above error message popped up in a MCO cluster and is this an expected behavior in this type of scenario?
>
Not expected in any case. It's a bug.

**Comment by aruniiird** [comment]
> > Still seeing error messages, like the following,
> > ```
> > failed to get PVC appset1/busybox-pvc: persistentvolumeclaims "busybox-pvc" is forbidden:
> > User "system:serviceaccount:openshift-storage:ocs-metrics-exporter" cannot get resource "persistentvolumeclaims" in API group "" in the namespace "appset1"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This is an expected error message, as we are granting `ocs-metrics-exporter` permissions to watch only the namespace it is currently deployed (here it is `openshift-storage`).
> 
> This shouldn't happen. There's a bug somewhere.
> 
> > Question here is, above error message popped up in a MCO cluster and is this an expected behavior in this type of scenario?
> 
> Not expected in any case. It's a bug.

Currently unable to repro the error. Raised a separate issue https://github.com/red-hat-storage/ocs-operator/issues/2327 to track this

Attaching the current metrics exporter log

[ocs-metrics-exporter.logs.tar.gz](https://github.com/red-hat-storage/ocs-operator/files/13682205/ocs-metrics-exporter.logs.tar.gz)

File: metrics/internal/cache/pv.go [hunk 1]
```diff
@@ -94,8 +94,14 @@ func appendIfNotExists(slice []string, value string) []string {
...
-	pv, ok := obj.(*corev1.PersistentVolume)
-	if !ok {
+	var pv *corev1.PersistentVolume
+	// obj can be of PV type or a pointer to it
+	switch pvFromObj := obj.(type) {
+	case *corev1.PersistentVolume:
+		pv = pvFromObj
+	case corev1.PersistentVolume:
+		pv = &pvFromObj
+	default:
```
Comment (umangachapagain) [inline metrics/internal/cache/pv.go:104]: We never needed this before. Why do we need it now?
What change caused us to require this?

Comment (aruniiird) [reply metrics/internal/cache/pv.go:104]: We have a long standing BZ raised for this : https://bugzilla.redhat.com/show_bug.cgi?id=2208302
It seems like issue is there for some time

**Review by umangachapagain — APPROVED** [review]

**Review by umangachapagain — APPROVED** [review]

**Pull request merged**